### PR TITLE
Job states

### PR DESCRIPTION
**Pull request by fkaercher**
* Updated JobState enum, see also #23  
* The changes have to be discussed

### CODE REVIEW

File: src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy [hunk 1]
```diff
@@ -128,7 +128,7 @@ class BEIntegrationTestStarter {
             // The job abortion might be a valid command but the cluster system might still try to keep the jobs.
             // We cannot handle problems like a stuck cluster, so let's stick to the basic query and see, if the job
             // ends within some seconds.
-            ensureProperJobStates(maxSleep, jobList, [JobState.ABORTED, JobState.OK, JobState.COMPLETED_UNKNOWN, JobState.COMPLETED_UNKNOWN], jobManager)
+            ensureProperJobStates(maxSleep, jobList, [JobState.ABORTED, JobState.COMPLETED_UNKNOWN, JobState.COMPLETED_UNKNOWN], jobManager)
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy]: Why did you remove OK? You can also remove the double UNKNOWN. Only one is needed.

File: src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy [hunk 2]
```diff
@@ -167,7 +167,7 @@ class BEIntegrationTestStarter {
 
             log.always("Abort jobs.")
             jobManager.queryJobAbortion(testJobs)
-            ensureProperJobStates(maxSleep, testJobs, [JobState.ABORTED, JobState.OK, JobState.COMPLETED_UNKNOWN, JobState.COMPLETED_UNKNOWN], jobManager)
+            ensureProperJobStates(maxSleep, testJobs, [JobState.ABORTED, JobState.COMPLETED_UNKNOWN, JobState.COMPLETED_UNKNOWN], jobManager)
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy]: dto

File: src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy (hunk 1 above)
Comment (fkaercher) [inline src/main/groovy/de/dkfz/roddy/BEIntegrationTestStarter.groovy]: I removed OK because it wasn't used any more. Also the meaning of status "OK" is not clear.

File: src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSJobManager.groovy [hunk 3]
```diff
@@ -688,4 +686,22 @@ class PBSJobManager extends ClusterJobManager<PBSCommand> {
         return PBSCommand.QSUB
     }
 
+
+    static JobState parseJobState(String stateString) {
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSJobManager.groovy]: You could un-static this method and make it abstract. This way it would be enforced to implement it.

File: src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSQstatReader.groovy [hunk 4]
```diff
@@ -109,7 +109,7 @@ class PBSQstatReader {
             gj.setCpuTime(jobResult.get("resources_used.cput") ? Duration.parse("PT" + jobResult.get("resources_used.cput").substring(0, 2) + "H" + jobResult.get("resources_used.cput").substring(3, 5) + "M" + jobResult.get("resources_used.cput").substring(6) + "S") : null)
             gj.setServer(jobResult.get("server"))
             gj.setUmask(jobResult.get("umask"))
-            gj.setJobState(JobState.parseJobState(jobResult.get("job_state")))
+            gj.setJobState(PBSJobManager.parseJobState(jobResult.get("job_state")))
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSQstatReader.groovy]: Then you could just call the parent job managers parse method.

File: src/main/groovy/de/dkfz/roddy/execution/jobs/JobState.java [hunk 5]
```diff
@@ -12,57 +12,43 @@
...
      */
     COMPLETED_UNKNOWN,
     /**
-     * BEJob jobState is unknown
-     */
-    UNKNOWN,
-    /**
-     * BEJob jobState is unknown because it was freshly read out from a file.
-     */
-    UNKNOWN_READOUT,
```
Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/JobState.java]: Did you check the Roddy sources, if some of the states are used?

Comment (fkaercher) [inline src/main/groovy/de/dkfz/roddy/execution/jobs/JobState.java]: * "OK" is only used in the ExecutionService
* "Unstarted"  is used -> I would keep it
* "Started" is used to check JobStatus against it in ExecutionContextReaderAndWriter, but "Started" is nowhere set
* "Unknown_Readout","Unknown_Submitted","Failed_Possible" are never used

File: src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSJobManager.groovy (hunk 3 above)
Comment (fkaercher) [inline src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSJobManager.groovy]: I would change it to private, because it's an internal method

Comment (dankwart-de) [inline src/main/groovy/de/dkfz/roddy/execution/cluster/pbs/PBSJobManager.groovy]: protected is ok
